main.go

```diff
@@ -16,11 +16,117 @@
 package main
 
 import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"log"
 	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-openapi/runtime"
+	"github.com/google/go-github/v42/github"
+	"github.com/google/trillian/merkle/logverifier"
+	"github.com/google/trillian/merkle/rfc6962"
+	"github.com/rhysd/actionlint"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/cosign/bundle"
+	"github.com/sigstore/cosign/pkg/cosign/tuf"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/client/entries"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	hashedrekord "github.com/sigstore/rekor/pkg/types/hashedrekord/v0.0.1"
+	rekord "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
 )
 
+// verifySignature receives the scorecard analysis payload, looks up its associated tlog entry and
+// certificate, and extracts the repository's workflow file to ensure its legitimacy.
+func verifySignature(w http.ResponseWriter, r *http.Request) {
```

Comment:
this function should return an `error`

Reply:
`http.HandleFunc` only handles functions with no return value though

Reply:
I see, did not know about that :-) Pretty weird